chore(weave): Fixes bugs related to path and trace tree params

tssweeney opened the pull request:
This PR does a few things:
1. Migrates the URL search params (`tracetree` and `path`) that are used in the call page to control trace visibility and selected subpath (respectively) to use the standard router to manage their state. As a result, we: a) can simplify a number of workarounds related to this; b) this fixes the issue where if you navigate from a trace tree, to an object, then click expand, the page breaks.
2. Modify the logic such that we only preserve subpath selection when we are navigating between calls to `Evaluate-predict_and_score`. This is a specific heuristic as it is the only case we actually want this behavior
3. Change the loading order of the call page such that we only block on loading the whole trace when needed - giving us speed in most cases, but allow us to correctly retain the subpath when needed.

jamie-rasmussen reviewed (commented):
This PR seems to introduce some changes and I'm not sure if they were intended:
* The tracetree seems to default to hidden for all calls, not just Evaluation-evaluate
* If I have navigated to a descendent and close the trace tree I revert to viewing the root.

tssweeney commented on weave-js/src/components/PagePanelComponents/Home/Browse2/RunsTable.tsx:
```diff
@@ -335,6 +338,14 @@ export const RunsTable: FC<{
     }
   }, [rowIds, peekId]);
 
+  // Custom logic to control path reservation preference
```
👍

tssweeney commented on weave-js/src/components/PagePanelComponents/Home/Browse3/context.tsx:
```diff
@@ -548,30 +558,9 @@ const useSetSearchParam = () => {
-const TRACETREE_PARAM = 'tracetree';
-const PEAK_SEARCH_PARAM = 'peekPath';
-const PATH_PARAM = 'path';
+export const PEAK_SEARCH_PARAM = 'peekPath';
```
👍

tssweeney commented:
> This PR seems to introduce some changes and I'm not sure if they were intended:
> 
> * The tracetree seems to default to hidden for all calls, not just Evaluation-evaluate
* Fixed! Thank you

> * If I have navigated to a descendent and close the trace tree I revert to viewing the root.
* Fixed! Thank you
